lib/internal/test_runner/reporter_stream.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+const {
+  ArrayPrototypePush,
+  ArrayPrototypeShift,
+} = primordials;
+const Readable = require('internal/streams/readable');
+
+class ReporterStream extends Readable {
+  #buffer;
+  #canPush;
+
+  constructor() {
+    super({ objectMode: true });
+    this.#buffer = [];
+    this.#canPush = true;
+  }
+
+  _read() {
+    this.#canPush = true;
+
+    while (this.#buffer.length > 0) {
+      const chunk = ArrayPrototypeShift(this.#buffer);
```

Comment:
I would call this something other than `chunk` since the stream is in object mode.